docs/dependency-injection/dependency-injection.md

```diff
@@ -0,0 +1,228 @@
+# What is Dependency Injection (DI)?
+
+Dependency injection (DI) is a design pattern that provides loosely-coupled, maintainable, and testable code. It improves modularity, maintainability, testability, and is easily customizable and extensible.
+
+In the context of Oryx, dependency injection allows users to customize logic deep inside the framework, which is particularly useful for projects with complex or rapidly-evolving requirements. Without DI, users would need to override large portions of the logic or create a lot of boilerplate code. By leveraging DI, users can override logic while still being able to upgrade to newer versions of Oryx going forward.
+
+One key advantage of using Oryx's DI implementation is that it is vanilla JavaScript and can be used in other frameworks as well. Although there are popular DI packages available, using Oryx's implementation ensures seamless integration and compatibility with its features.
+
+# Service Layer in Oryx
+
+The service layer in Oryx serves as the foundation for the business logic, with dependency injection (DI) playing a crucial role.
+
+The main objective of the service layer in Oryx is to abstract all of the system functionality, including querying backend services, data caching and reloading, state management, and reactivity.
+
+Furthermore, DI is utilized by many higher-level elements and concepts in Oryx, ranging from adapters and normalizers to HTTP interceptors, authentication, product loading, cart management, and the checkout process.
+
+## Npm package
+
+The `@spryker-oryx/di` package provides dependency injection not only to Oryx applications but also to other packages and can even be used outside the Oryx framework. It includes a variety of useful utilities to streamline working with dependency injection, such as:
+
+- TypeScript models definitions (e.g. Providers)
+- `Injector` - the implementation of a DI container in TypeScript
+- Functions used to set up and manage injectors (`createInjector()`, `getInjector()`, etc.)
+- Utilities that can be used to resolve dependencies (`inject` and `resolve`)
+
+In a typical Oryx application, the application orchestrator automatically handles much of the work related to setting up the DI container. Nevertheless, the `@spryker-oryx/di` library provides a set of tools that can be used to further customize and fine-tune the dependency injection as needed.
+
+# Using services
+
+## inject vs resolve
+
+There are two primary methods for injecting services and dependencies:
+
+- `inject` – usually used inside services
+- `resolve` – usually used inside (web) components
```

Comment:
usually? What other cases would we have? Can we drop it to not overcomplicate it?